### PR TITLE
[DOCS] Revise CCR and snapshot compatibility

### LINKED ISSUE #83628: [DOCS] Clarify CCS/CCR version compatibility
cc @lockewritesdocs and @giladgal 

It has been pointed out https://github.com/elastic/kibana/issues/124465#issuecomment-1028516829 that our documentation around CCS/CCR version compatibility doc may not be entirely accurate.

I have solicited some of the feedback around the following pages, and we might want to re-architecture the content to make sure they are informative and consistent with the supportability.

### Pages to be updated:
- https://www.elastic.co/guide/en/elasticsearch/reference/master/remote-clusters.html
- https://www.elastic.co/guide/en/elasticsearch/reference/master/xpack-ccr.html
- https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-supported-configurations
- https://www.elastic.co/guide/en/cloud/current/ec-enable-ccs.html (Elastic Cloud page)

### Issues:
1. On the [remote-clusters](https://www.elastic.co/guide/en/elasticsearch/reference/master/remote-clusters.html) page, two `Version compatibility table` are present.  One under `sniff mode`, one under `proxy mode`.
As far as I can see, these two tables are identical and do not serve the purpose for either CCS nor CCR compatibility.
    - This can be super confusing to users because we actually have 3 sets of rules, one for CCS, one for CCR, and one for remote cluster connectivity.  The latter doesn't really serve any purpose other than the clusters can be connected, but individual features supportability constraints are listed elsewhere.  See items 3 and 4.

2. On the [xpack-ccr](https://www.elastic.co/guide/en/elasticsearch/reference/master/xpack-ccr.html) page, CCR compatibility should be explicitly pointed out that it relies on both cluster version and _index_ version.  
    - The latter is currently missing from the doc.

3. For CCR cluster version compatibility is stated as follows:
    - In a uni-directional configuration, the cluster containing follower indices must be running the same or newer version of Elasticsearch as the remote cluster. If newer, the versions must also be compatible as outlined in the following matrix.
      - `Version compatibility matrix` (which is the same table as the one in the [remote-cluster](https://www.elastic.co/guide/en/elasticsearch/reference/master/remote-clusters.html) page, yet the table does not conform with the above rules. 
    - In a bi-directional configuration, version compatibility is not stated on the doc. 

4. For CCS cluster version compatibility is stated as follows:
    - Elastic only supports searches from a local cluster to a remote cluster running:
      - The previous minor version.
      - The same version.
      - A newer version. This version must also be compatible with the local cluster as outlined in the following matrix.
        - `Version compatibility matrix` (which is the same table as the one in the [remote-cluster](https://www.elastic.co/guide/en/elasticsearch/reference/master/remote-clusters.html) page, yet the table does not conform with the above rules. 

5.  Once all these pages have been updated, there should be a cloud doc update linking to the corresponding pages.

### Proposal
- [x] P1. We should also probably spell out explicitly in the [docs on remote clusters](https://www.elastic.co/guide/en/elasticsearch/reference/master/remote-clusters.html) that we’re just talking about connectivity between clusters there, and individual cross-cluster features may impose stricter compatibility constraints.   Since the connectivity doesn't really serve any real purpose, I propose to remove both `Version compatibility table` from the [remote clusters](https://www.elastic.co/guide/en/elasticsearch/reference/master/remote-clusters.html) page and redirect users to [CCS](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-supported-configurations) and [CCR](https://www.elastic.co/guide/en/elasticsearch/reference/master/xpack-ccr.html) compatibility pages to avoid duplicated information scattered.

Instead of the following:
```
Important!
Elastic only supports cross-cluster search on a subset of these configurations. See [Supported cross-cluster search configurations](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-supported-configurations).
```
Perhaps:
```
Important!
Individual cross-cluster features may impose stricter compatibility constraints.  See [Supported Cross-cluster search configurations](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-supported-configurations) and [Supported Cross-cluster replication configurations](https://www.elastic.co/guide/en/elasticsearch/reference/master/xpack-ccr.html)
```
- [x] P2.  CCR also requires index compatibility which is similar to what we have for the snapshot today: [snapshot index compatibility](https://github.com/elastic/elasticsearch/edit/7.17/docs/reference/snapshot-restore/index.asciidoc)
Perhaps:
```
Any follower index you replicate from leader must also be compatible with the current cluster’s version. If you try to replicate an index created in an incompatible version, the replication attempt will fail. 
```
And we can also add an index compatibility table like this (It's not clear to me whether this applies to both uni- and bi-directional??):
<details>
      <summary> Click to expand the CCR index version compatibility table (⚠️ TBC) </summary>

Screen Shot 2022-02-08 at 3 10 40 pm
    </details>
    
- [x] P3.1 As you can see, the CCR uni-directional cluster version compatibility table gets a little bit complicated with all the clauses.  I am happy to remove the table completely if we can state them in some words that make more sense.  If not, we can continue to have the table, but we want to avoid keep adding new versions to the table for maintainability. 
    - the following table removes some of the old 5.x and early 6.x EOL versions.
    - the following table follows the above rule where the follower cluster must be running the same or newer version as the leader.
    - the following table adds an asterisk to all the diagonal element to indicate all the versions in the ranges like `7.1-7.16` still needs to obey the rule where the follower must be running the same or newer version as the leader.
    - It is also much more clear to state `follower/leader` vs `remote/local` cluster for the purpose of CCR.
    - If we decided to keep the table and update the content, can we keep the table uncollapsed by default? This is to ensure the table is visible when users navigate to the page.  There have been several occasions the table has been overlooked.
<details>
  <summary> Click to expand the CCR uni-directional cluster version compatibility table</summary>

  Screen Shot 2022-02-07 at 6 09 00 pm
</details>
    
- [x] P3.2 For CCR bi-directional, we don't seem to have any version rule explicitly mentioned on the [xpack-ccr](https://www.elastic.co/guide/en/elasticsearch/reference/master/xpack-ccr.html) page.  Can we add something like this to the effect? 
```
the two clusters in bi-directional replication must remain on the same version
```

- [x] P4. CCS cluster version compatibility 
  - Since the table on the [CCS page](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-supported-configurations) does not conform with the stated rules, we might want to either remove the table or update the table like the following: (⚠️ TBC) 
  - If we decided to keep the table and update the content, can we keep the table uncollapsed by default? 
<details>
  <summary> Click to expand the CCS cluster version compatibility table</summary>

  Screen Shot 2022-02-08 at 2 37 13 pm 
</details>

- [x] P5.  Please follow up with a cloud doc update on this page. https://www.elastic.co/guide/en/cloud/current/ec-enable-ccs.html 
  - I think pointing directly to the CCS and CCR compatibility page is the best. 

----
This is a followup on https://github.com/elastic/elasticsearch/pull/81239

### PR DESCRIPTION
**Pull request by Leaf-Lin**
As per https://github.com/elastic/elasticsearch/issues/83628, I've created some changes around version compatibility.
### CCR
- Remove compatibility table from remote cluster page. Redirect users to [CCS](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-supported-configurations) and [CCR](https://www.elastic.co/guide/en/elasticsearch/reference/master/xpack-ccr.html) compatibility pages to avoid duplicated information scattered.
- Add bi-directional version compatibility
- Add uni-directional cluster version compatibility, added a new table which is different from remote cluster table.
- Add CCR index vs cluster version compatibility

### Snapshots
- Fix 7.0 vs 7.1 in snapshot/CCR index vs cluster version compatibility

Closes: https://github.com/elastic/elasticsearch/issues/83628

## Previews
* Remote clusters: https://elasticsearch_83916.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/remote-clusters.html#sniff-mode
* CCR compatibility: https://elasticsearch_83916.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/xpack-ccr.html#ccr-cluster-compatibility
* Snapshot/restore compatibility: https://elasticsearch_83916.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshot-restore.html#snapshot-cluster-compatibility

### CODE REVIEW

File: docs/reference/modules/remote-clusters-shared.asciidoc [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-| 5.6                 | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}
-| 6.0–6.6             | {no-icon}  | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}
-| 6.7                 | {no-icon}  | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {no-icon}  | {no-icon}  | {no-icon}
-| 6.8                 | {no-icon}  | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {no-icon}
-| 7.0                 | {no-icon}  | {no-icon}  | {no-icon}  | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {no-icon}
-| 7.1–7.16            | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}  | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {no-icon}
-| 7.17                | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}  | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon} | {yes-icon}
-| 8.0–{minor-version} | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}  | {no-icon}  | {yes-icon} | {yes-icon}
```
Comment (DaveCTurner) [inline docs/reference/modules/remote-clusters-shared.asciidoc:14]: Not sure we need to mention 6.7 and earlier, nor 7.0 (these versions are all ≥1 year past EOL) but we do need to have an `8.0–{minor-version}` group.

File: docs/reference/snapshot-restore/cluster-index-compat.asciidoc [hunk 2]
```diff
@@ -2,11 +2,9 @@
...
-| 6.0–6.7                   | {yes-icon} | {yes-icon} | {yes-icon}            | {no-icon}
-| 6.8                       | {yes-icon} | {no-icon}  | {yes-icon}            | {no-icon}
-| 7.0–7.1                   | {no-icon}  | {yes-icon} | {yes-icon}            | {yes-icon}
-| 7.2–{prev-major-last}     | {no-icon}  | {no-icon}  | {yes-icon}            | {yes-icon}
-| 8.0–{minor-version}       | {no-icon}  | {no-icon}  | {no-icon}             | {yes-icon}
+h| Index creation version   | 6.8        | 7.0       |  7.1–{prev-major-last} | 8.0–{minor-version}
+| 6.8                       | {yes-icon} | {no-icon} |  {yes-icon}            | {no-icon}
+| 7.0                       | {no-icon}  | {yes-icon}|  {yes-icon}            | {yes-icon}
+| 7.1–{prev-major-last}     | {no-icon}  | {no-icon} |  {yes-icon}            | {yes-icon}
+| 8.0–{minor-version}       | {no-icon}  | {no-icon} |  {no-icon}             | {yes-icon}
```
Comment (DaveCTurner) [inline docs/reference/snapshot-restore/cluster-index-compat.asciidoc:9]: Likewise here, I think we should drop the 7.0 entries.

**Review by DaveCTurner — COMMENTED** [review]
I'd need to see the preview to review this properly but I left a couple of comments about which versions we mention anyway. I will raise this for discussion in a wider group too.

**Review by jrodewig — COMMENTED** [review]
Thanks again @Leaf-Lin. I pushed a few commits to address David's feedback, fix the docs build, and split off the CCS table.

With those changes, this looks good to me. However, since I've contributed, I'm probably no longer a good reviewer.

I also left some optional suggestions for the CCR uni-directional table that I'd like some feedback on before commiting.

**Comment by jrodewig** [comment]
We updated the CCS compatibility table as part of https://github.com/elastic/elasticsearch/pull/84055 to accommodate some additional changes to our FWC strategy for CCS. As a result, I've removed the CCS-related changes from this PR.

@DaveCTurner @Leaf-Lin Let me know if any other changes are needed. Once this is approved, I can get the changes merged in and backported.

**Comment by jrodewig** [comment]
@DaveCTurner @Leaf-Lin  I just wanted to check in and see you have any feedback for this PR. Thanks again!

**Comment by debadair** [comment]
@elasticmachine Retest this please

**Comment by Leaf-Lin** [comment]
We discussed this and felt that the doc should be focusing compatibility matrices just on the current version of the docs and stating the compatibility rules, while linking to older versions for their respective information.

**Pull request closed without merging**